app/models/grda_warehouse/tasks/identify_duplicates.rb

```diff
@@ -115,111 +131,511 @@ def identify_duplicates
+      end
+      exact_name_matches.each do |id_pair|
+        matches[id_pair] ||= 0
+        matches[id_pair] += 1
+      end
+      exact_dob_matches.each do |id_pair|
+        matches[id_pair] ||= 0
+        matches[id_pair] += 1
+      end
+      more_than_one_match = matches.select { |_, v| v > 1 }
+
+      # Get split history
       all_splits = GrdaWarehouse::ClientSplitHistory.pluck(:split_from, :split_into)
-      splits_by_from = all_splits.group_by(&:first)
-      splits_by_into = all_splits.group_by(&:last)
-
-      @source_clients.each do |target|
-        splits = splits_by_from[target.id]&.flatten || [] # Don't re-merge anybody that was split off from this candidate
-        splits += splits_by_into[target.id]&.flatten || [] # Don't merge with anybody that this candidate was split off from
-
-        matches_name = @source_name_lookup.get_ids(first_name: target.first_name, last_name: target.last_name)
-        matches_ssn = @source_ssn_lookup.get_ids(ssn: target.ssn)
-        matches_dob = @source_dob_lookup.get_ids(dob: target.dob)
-        all_matching_dest_ids = (matches_name + matches_ssn + matches_dob) - splits
-        all_matching_dest_ids.filter! { |id| id != target.id }
-        # to_merge_by_dest_id = Set.new
-        # seen = Set.new
-        # all_matching_dest_ids.each do |num|
-        #   if seen.include?(num)
-        #     to_merge_by_dest_id << num
-        #   else
-        #     seen << num
-        #   end
-        # end
-        to_merge_by_dest_id = all_matching_dest_ids.uniq.
-          map { |num| [num, all_matching_dest_ids.count(num)] }.to_h.
-          select { |_, v| v > 1 }
-
-        to_merge += to_merge_by_dest_id.keys.map { |source_id| [target.id, source_id].sort } if to_merge_by_dest_id.any?
+      @logger.info "Found #{all_splits.size} split history records: #{all_splits.inspect}"
+
+      # Filter more_than_one_match removing any that were split previously
+      all_splits.each do |row|
+        sorted_row = row.sort
+        next unless more_than_one_match.key?(sorted_row)
+
+        @logger.info "Removing previously split pair: #{sorted_row.inspect}"
+        more_than_one_match.delete(sorted_row)
+      end
+      candidates = group_merge_chains(more_than_one_match)
+
+      source_client_counts = GrdaWarehouse::WarehouseClient.group(:destination_id).count
+      split_chains_on_max_source(candidates: candidates, counts: source_client_counts)
+    end
+
+    # Find potential client merge candidates by looking for matches across multiple criteria
+    # @return [Hash] A set of client ID pairs that are candidates for merging
+    # @note Only returns matches if auto deduplication is enabled in config
+    private def find_merge_candidates_for_unprocessed
+      # Never return obvious matches if auto deduplication is disabled
+      unless GrdaWarehouse::Config.get(:enable_auto_deduplication)
+        @logger.info 'Auto deduplication disabled, returning empty set'
+        return {}
```

Comment:
always return an array
```suggestion
        return []
```